Sources/Defaults/Observation.swift

```diff
@@ -216,17 +169,20 @@ extension Defaults {
 		}
 
 		func start(options: ObservationOptions) {
-			object?.addObserver(self, forKeyPath: key, options: options.toNSKeyValueObservingOptions, context: nil)
+			guard !isObserving else {
+				return
+			}
+			suite?.addObserver(self, forKeyPath: name, options: options.toNSKeyValueObservingOptions, context: &Self.observationContext)
 			isObserving = true
 		}
 
 		func invalidate() {
-			if isObserving {
-				object?.removeObserver(self, forKeyPath: key, context: nil)
-				isObserving = false
+			guard isObserving else {
+				return
 			}
-
-			object = nil
+			suite?.removeObserver(self, forKeyPath: name)
+			isObserving = false
+			suite = nil
 		}
```

Comment:
Unrelated to this specific pull request, but I wonder if this is thread-safe. It may be smart to put a lock on it?

Reply:
Put a lock on it. Maybe we should consider to use AtomicBoolean after swift 6?

Reply:
> Maybe we should consider to use AtomicBoolean after swift 6?

👍